client/src/Component/views/Methods/index.js

```diff
@@ -48,79 +51,134 @@ class Methods extends Component {
+  removeMethod = (points, event) => {
     let { resources } = this.state;
     this.setState(nextState => {
       resources = nextState.resources + points;
       return { resources };
     });
   };
 
-  // checks remaining resources and triggers redirect to error page if you've over spent
-  errorOverSpend = () => {
-    const { resources } = this.state;
-    if (resources < 0) {
-      this.setState({
-        redirect: true,
-      });
-    }
+  // Triggers if you try and overspend your resources
+  noMoreResources = () => {
+    Swal.fire({
+      type: 'error',
+      title: 'Oops!',
+      text: "You don't have enough resources!",
+      confirmButtonText: 'Go back and edit your selection.',
+      confirmButtonColor: '#faa634',
+      showCancelButton: true,
+      cancelButtonText: "Let's move on.",
+    }).then(value => {
+      const { history } = this.props;
+      if (value.dismiss === 'cancel') {
+        history.push('/priorities');
+      } else if (value.value === true) {
+        history.push('/methods');
+      }
+    });
   };
 
-  // renders redirect if there are no resources left
-  renderRedirect = () => {
-    const { redirect } = this.state;
-    if (redirect) {
-      return <Redirect to="/errorNoMoreResources" />;
+  // Triggers alerts on next click
+  nextAlerts = (selectedCards, resources) => {
+    if (selectedCards.length === 0) {
+      Swal.fire({
+        type: 'error',
+        title: 'Oops...',
+        text: 'You must select your methods first!',
+        confirmButtonText: 'OK',
+        confirmButtonColor: '#faa634',
+      }).then(() => {
+        const { history } = this.props;
+        history.push('/methods');
+      });
+    } else if (resources > 0) {
+      Swal.fire({
+        type: 'info',
+        title: 'Are you sure you want to move forward?',
+        text: 'You still have resources left to spend!',
+        confirmButtonText: 'Go back and spend more resources.',
+        confirmButtonColor: '#faa634',
+        showCancelButton: true,
+        cancelButtonText: "Let's move on.",
+      }).then(value => {
+        const { history } = this.props;
+        if (value.dismiss === 'cancel') {
+          history.push('/priorities');
+        } else if (value.value === true) {
+          history.push('/methods');
+        }
+      });
     }
-    return null;
   };
 
   render() {
     const { resources } = this.state;
     return (
-      <Fragment>
-        {this.renderRedirect()}
-        <Header headerImg={null} titleText="Select your methods" />
-        <Carousel
-          wrapAround
-          enableKeyboardControls
-          renderCenterLeftControls={({ previousSlide }) => (
-            <PrevBtn onClick={previousSlide}>.</PrevBtn>
-          )}
-          renderCenterRightControls={({ nextSlide }) => (
-            <NextBtn onClick={nextSlide}>.</NextBtn>
-          )}
-        >
-          {this.renderMethodCards()}
-        </Carousel>
-        <Line />
+      <MyContext.Consumer>
```

Comment:
`Methods.contextType = MyContext;`
is declered, so u don't need this block 
```jsx
<MyContext.Consumer>
     {context => { .... }
</MyContext.Consumer>
```
u just add 
```jsx
render() {
    const { resources } = this.state;
    const { selectedCards } = this.context;
```